R/stack.R

```diff
@@ -0,0 +1,76 @@
+#' R6 Class for a FIFO stack
+#'
+#' @description
+#'
+#' This is a simple stack object offering add / pop functionality
+#'
+#' @import R6
+#' @export
```

Comment:
Just to understand, why did you choose a R6 class for this, and not a S3 method for jackknife and bootstrap?

Reply:
I used R6 for the stack object itself as it allows for in-place modification whilst still being able to return a value

```
x <- Stack()
val1 <- x$pop()
val2 <- x$pop()
```

vs something like

```
x <- Stack()

z1 <- pop(x)
x <- z1$x
val1 <- z1$vals

z2 <- pop(x)
x <- z2$x
val2 <- z2$vals
```

In terms of implementing the jackjnife and bootstrap I probabally could put an S3 wrapper around the Stack object which is pretty much what I've done [here](https://github.com/insightsengineering/rbmi/pull/276/files#diff-e26e6ab8b4b9c6e3658ed099e58fa294d8e8d7afa29f709ed5a4c66bcbefc5a0R49). In the end I opted not to because the bootstrap and condmean both have the same implementation so would need to have a 3rd function for them both to point to and it just got a bit messy.